daffodil-sapi/src/main/scala/org/apache/daffodil/sapi/Daffodil.scala

```diff
@@ -1089,3 +1090,18 @@ class DaffodilUnparseContentHandler private[sapi] (sContentHandler: SDaffodilUnp
   override def skippedEntity(name: String): Unit =
     contentHandler.skippedEntity(name)
 }
+
+/**
+ * Exposes access to the DFDLCatalogResolver to the Java API.
+ *
+ * Resolves URI/URL/URNs to loadable files/streams.
+ *
+ * The DFDLCatalogResolver implements the following interfaces:
+ *  - org.apache.xerces.xni.parser.XMLEntityResolver
+ *  - org.w3c.dom.ls.LSResourceResolver
+ *  - org.apache.sax.EntityResolver
+ *  - org.apache.sax.ext.EntityResolver2
+ */
```

Comment:
I think we need a blurb about thread safety. Something about the returned EntityResolver not being thread safe, but if this get function is called from within different threads then it is since it returns a unique `EntityResolver` per thread instance. I'm not sure the best way to word that, but something like that feels important.

And thinking about threads, I wonder if we should always return a new instance, and we just say it's not thread safe? If the user want to make optimizations like what Daffodil does and store instances in a ThreadLocal then they can do that themselves. I'm not sure we should be doing that for them and forcing them into a single instance per thread?